### ISSUE TITLE
We need to clean up and organize the issues

### ISSUE THREAD

**Issue by craftworkgames** [craftworkgames/MonoGame.Extended#187]
The github issues system is great in many ways.

However, when there's lots of open issues that have no clear goal it becomes difficult to manage. Let's discuss what we can do about that.

What should we be using issues for and what shouldn't we?
How can we clean up the existing pile?

**Comment by ExCellRaD** [comment]
I think there shouldn't be issues for open questions such as #184 and #167. They don't get much traction and don't get closed, they should probably only be discussed in other topics which are relevant to them. 
Maybe even remove suggestions such as #180 and #11 and try to make them more organized. A good way would be to point to the monogame forums for that. Keeping them in a separate thread and sorting them on importance. Having some kind of a direction for implementing features would be great and much more efficient. There is a lot of discussion going on in these issues right now, but they don't seem to get closed. Feature implementation discussion could also be moved to the PR's instead, which means only the ones that are getting worked on are there.
Issues can then be kept only for things that need fixing and bugs, and could then be closed much faster.

**Comment by LithiumToast** [comment]
I would like to see issues getting some more tags so they can be sorted quickly. Something like https://robinpowered.com/blog/best-practice-system-for-organizing-and-tagging-github-issues/

**Comment by craftworkgames** [comment]
> I think there shouldn't be issues for open questions such as #184 and #167. They don't get much traction and don't get closed, they should probably only be discussed in other topics which are relevant to them.

Actually, those 2 issues are easy deal with because they have a well defined outcome.
 - Should we use Game Components?  Yes or no. We either decide to do it or we don't. Closed.
 - Should FastRand be part of the core library? Yes or no. 

Of course, it's often not a simple yes or no answer because we need to discuss *how* to do it as well. API design is very important. We need somewhere to discuss it and I think an issue is a sensible place.

The only problem I see with these two issues is *when* to close them. Should we close them as soon as a decision has been made, or after the implementation has been committed? 

> Maybe even remove suggestions such as #180 and #11 and try to make them more organized.

#180 is a new issue. I've barely had a chance to look at it. I take your point though and I will try to make a decision on these types of issues sooner rather than leaving them lingering around.

#11 is definitely an old issue. It's a candidate for being closed. I'm just not sure if we should capture the relevant bits into new issues, or just close it.

> A good way would be to point to the monogame forums for that.

Some issues might be better on the forums perhaps. It depends how we define what an issue is. To me, the distinction is about the audience reading it:
 - github issues are targeted at the developers working on the code.
 - the forums are targeted at the users of the library.

> Having some kind of a direction for implementing features would be great and much more efficient.

This is something I agree with 100%. We definitely need a clearly defined list of features that should be in the library before we reach version 1.0. I've been thinking about this a lot lately and I'm putting something together. The only question is, where should it be published?

> There is a lot of discussion going on in these issues right now, but they don't seem to get closed.

Yep. I think each issue needs a clear objective. Some are obvious, like bug fixes, but others are less clear. That's the point of this discussion. We are trying to establish a process for closing issues.

> Feature implementation discussion could also be moved to the PR's instead, which means only the ones that are getting worked on are there.

Maybe. The problem with this approach is that some things need a discussion *before* we get to the PR. In fact, I'd say jumping into the PR to early creates a problem rather than resolving one. 

Trying to design the API at the PR level doesn't really work either. We've tried that a few times and it's a big waste of time and effort for everyone involved (usually myself and the other person). Having a rough idea what the API looks like *before* writing any code seems to be a better way to go.

> Issues can then be kept only for things that need fixing and bugs, and could then be closed much faster.

How would we enforce that? People are going to raise issues for whatever reason they feel like. Documentation might help, but at the end of the day we still need a place to discuss things and as I said above, I don't think the forums or PR's are the right place.

> I would like to see issues getting some more tags

Good idea. More tags should help in theory. Let's start with a sensible subset of the ones in that blog post and see how it goes.

**Comment by guFalcon** [comment]
My two cents:

I think we should *discuss* things on a *discussion board* and not on an *issue tracker*.
We have a beautiful board over at MonoGame; We should use that for topics like 'Should we add this or that to the library?'. The upside would be that more people could join in.
Topics like 'should we >*extremely difficult technical blah here*<?' should probably stay here.

For now:
On the 'decision issues' I think we should really close them as soon as a decision is reached; like we did with the per-pixel-collision topic. No need to keep them lingering.
As soon as you post something along the lines of 'we don't add that right now' I will close those.
I'm not so picky when it comes to closing issues. They're not gone.

On 'defined list of features and where to put it' I would add a paragraph in the main-readme.md file called 'Roadmap' or something like that.
The 'Design goals' section already addresses the vision of the project, so that's perfectly fine.

Rest sounds perfect.
The tags are nice.

Just my two cents.

**Comment by craftworkgames** [comment]
I've just added a few new tags and applied them to all current issues. There's room for more as needed.

> On the 'decision issues' I think we should really close them as soon as a decision is reached;

I'm okay with this as long as we have another place to capture the intent to implement. In a company this would typically be some sort of task board or backlog. 

I currently keep a private [Trello board](https://trello.com/) to stay organized but it might help to have a public one. I tried this early in the project. Unfortunately it only works well if everyone uses it.

> I'm not so picky when it comes to closing issues. They're not gone.

The only real problem with closing issues is that they get forgotten. That's why I'd like to capture them somewhere else if we go down this path.

**Comment by craftworkgames** [comment]
> On 'defined list of features and where to put it' I would add a paragraph in the main-readme.md file called 'Roadmap' or something like that.

That seems like the most visible place. I did start one [on the wiki](https://github.com/craftworkgames/MonoGame.Extended/wiki/MonoGame.Extended-Roadmap) a while ago but it's out of date and not very visible. 

I'd like to avoid having *every possible feature* on the roadmap. Let's limit it to a set of features that make up version 1.0 and things we've made a solid commitment to implement. Everything else stays as a feature request.
